tests/benchmark/bm_spaces.cpp

```diff
@@ -0,0 +1,288 @@
+#include <benchmark/benchmark.h>
+#include <random>
+#include <unistd.h>
+#include "VecSim/utils/arr_cpp.h"
+#include "VecSim/spaces/space_includes.h"
+#include "VecSim/spaces/space_interface.h"
+#include "VecSim/spaces/space_aux.h"
+
+#include "VecSim/spaces/L2/L2.h"
+#include "VecSim/spaces/IP/IP.h"
+
+class BM_VecSimSpaces : public benchmark::Fixture {
+protected:
+    std::mt19937 rng;
+    size_t dim;
+    float *v1, *v2;
+    Arch_Optimization opt;
+
+#define ASSERT_OPT(OPT)                                                                            \
+    if (1) {                                                                                       \
```

Comment:
Why `if(1)` ?

Reply:
`if(1) {<code>} else` and `do {<code>} while(0)` are two common ways to encapsulate a macro and make it that it can be treated as a function call, with adding `;` at the end without breaking the code inside the macro